kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/SolverServiceBase.java

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.services.optaplanner;
+
+import org.kie.server.api.model.*;
+import org.kie.server.api.model.instance.SolverInstance;
+import org.kie.server.api.model.instance.SolverInstanceList;
+import org.kie.server.services.api.KieContainerInstance;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.impl.KieContainerInstanceImpl;
+import org.optaplanner.core.api.domain.solution.Solution;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.api.solver.SolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+
+/**
+ * Optaplanner solver service
+ */
+public class SolverServiceBase {
+
+    private static final Logger logger = LoggerFactory.getLogger( SolverServiceBase.class );
+    private final ExecutorService executor;
+
+    private KieServerRegistry context;
+    private Map<String, SolverInstanceContext> solvers = new ConcurrentHashMap<String, SolverInstanceContext>();
+
+    public SolverServiceBase(KieServerRegistry context, ExecutorService executorService) {
+        this.context = context;
+        this.executor = executorService;
+    }
+
+    public ServiceResponse<SolverInstanceList> getSolvers(String containerId) {
+        try {
+            List<SolverInstance> sl = getSolversForContainer( containerId );
+            return new ServiceResponse<SolverInstanceList>(
+                    ServiceResponse.ResponseType.SUCCESS,
+                    "Solvers list successfully retrieved from container '" + containerId + "'",
+                    new SolverInstanceList( sl ) );
+        } catch ( Exception e ) {
+            logger.error( "Error retrieving solvers list from container '" + containerId + "'", e );
```

Comment:
I wondering if we can do anything to do all these exception handling code through an aspect?
It's always the same boilerplate, making the code harder to read.
With CDI we might be able to add a simple interceptor for all these methods I think.